src/Infrastructure/Persistence/Doctrine/Repository/Regulation/LocationRepository.php

```diff
@@ -68,19 +66,9 @@ public function findAllForMapAsGeoJSON(
                 ? 'AND ro.end_date IS NOT NULL'
                 : '');
 
-        $upcomingOnly = $includeUpcomingRegulations && !$includePastRegulations;
-        $pastOnly = !$includeUpcomingRegulations && $includePastRegulations;
-        $presentOnly = !$includeUpcomingRegulations && !$includePastRegulations;
-
-        $regulationDatesWhereClause =
-            $upcomingOnly
-            ? 'AND (((ro.end_date >= :now OR ro.end_date IS NULL) AND ro.start_date <= :now) OR ro.start_date > :now)'
-            : ($pastOnly
-                ? 'AND (((ro.end_date >= :now OR ro.end_date IS NULL) AND ro.start_date <= :now) OR ro.end_date < :now)'
-                : ($presentOnly
-                    ? 'AND ((ro.end_date >= :now OR ro.end_date IS NULL) AND ro.start_date <= :now)'
-                    : ''));
-
+        $noEntry = $includeMeasureTypeNoEntry && !$includeMeasureTypeSpeedLimitation;
+        $speedLimitation = $includeMeasureTypeSpeedLimitation && !$includeMeasureTypeNoEntry;
+        $regulationType = $noEntry ? 'AND m.type = \'noEntry\'' : ($speedLimitation ? 'AND m.max_speed IS NOT NULL' : '');
```

Comment:
Et grâce à l'array on pourrait tout simplifier en une condition 'IN'

```suggestion
        $regulationTypeCondition = 'm.type IN (:measureTypes)';
```

Dans le cas où aucune checkbox de type de mesure n'est coché, je m'attendrais à ce qu'on ne renvoie rien donc ce IN en l'état me semble adapté (`m.type IN []` ça renvoie bien aucun résultat)